gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractRegionMap.java

```diff
@@ -895,11 +894,14 @@ public void copyRecoveredEntries(RegionMap rm, boolean entriesIncompatible) {
             tombstones.put(re.getVersionStamp().asVersionTag(), re);
           }
         }
+
         LocalRegion.regionPath.set(_getOwner().getFullPath());
          int valueSize  = _getOwner().calculateRegionEntryValueSize(re);
         _getOwner().calculateEntryOverhead(re);
-        //Always take the value size from recovery thread.
-        _getOwner().acquirePoolMemory(0, 0, true, null, false);
+        // Always take the value size from recovery thread.
+        if (!re.isTombstone()) {
+          _getOwner().acquirePoolMemory(0, 0, true, null, false);
+        }
```

Comment:
@rishitesh don't understand this change. Is the base entry overhead already accounted for elsewhere? If yes then this method should have "withEntryOverhead" as false. If not then it should be called for tombstone too.

Reply:
The base value is accounted in recovery thread when its not even in a entry form. Just after reading from disk  and before setting the entry to region, we account the numb bytes de serialized. See DiskEntry.setValueOnFaultIn. Hence while changing the map owner we just have to account for entry overhead.

Reply:
@rishitesh But then why this is not required for tombstones? The tombstone values will still occupy region entry and that overhead need to be accounted for.